python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -3319,3 +3319,31 @@ def test_constructor_properties():
     # Inorrect use of _constructor_expanddim (Raises for DataFrame)
     with pytest.raises(NotImplementedError):
         df._constructor_expanddim
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        [1, 2, 3, 4, 5],
+        [1, 2, None, 4, 5],
+        [1.0, 2.0, 3.0, 4.0, 5.0],
+        [1.0, 2.0, None, 4.0, 5.0],
+        ["a", "b", "c", "d", "e"],
+        ["a", "b", None, "d", "e"],
+        [None, None, None, None, None],
+        np.array(["1991-11-20", "2004-12-04"], dtype=np.datetime64),
+        np.array(["1991-11-20", None], dtype=np.datetime64),
+        np.array(
+            ["1991-11-20 05:15:00", "2004-12-04 10:00:00"], dtype=np.datetime64
+        ),
+        np.array(["1991-11-20 05:15:00", None], dtype=np.datetime64),
+    ],
```

Comment:
Can we add a few tests with some `NaN` and `NaT` values mixed in along with nulls as well?
I.e.
```
[1.0, None, 3.0, np.nan, 5.0],
["1991-11-20 05:15:00", None, np.datetime64('NaT')]
```

Reply:
added - figured out a lot of ways of NOT to put a NaT into a Series or pyarrow array along the way.